src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -1030,35 +1122,51 @@ private void doEncodeProtocolHeader(
 
             replyBudgetReserved += protocolHeader.sizeof() + replyPadding;
             doNetworkData(traceId, authorization, 0L, protocolHeader);
-            if (protocolId == SASL_PROTOCOL_ID)
-            {
-                doEncodeSaslMechanisms(traceId, authorization);
-            }
         }
 
-        private void doEncodeSaslMechanisms(
+        private void doEncodeSaslProtocolHeader(
+            int protocolId,
+            int major,
+            int minor,
+            int revision,
             long traceId,
             long authorization)
+        {
+            final AmqpProtocolHeaderFW protocolHeader = amqpProtocolHeaderRW
+                .wrap(writeBuffer, DataFW.FIELD_OFFSET_PAYLOAD, writeBuffer.capacity())
+                .name(n -> n.set("AMQP".getBytes(StandardCharsets.US_ASCII)))
+                .id(protocolId)
+                .major(major)
+                .minor(minor)
+                .revision(revision)
+                .build();
```

Comment:
Seems that this could be pre-created and shared at factory scope rather than rebuilding each time.